### PR TITLE
Add a README to the CLI distribution package

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose

Bundle a README and the bash completion script with the CLI distribution package

### CODE REVIEW

File: cmd/build.sh [hunk 1]
```diff
@@ -125,24 +125,35 @@ do
...
-    zipdir="${buildPath}/$filename"
-    mkdir -p $zipdir
+    mi_dir_name="wso2$filename-cli-$build_version"
+    mi_archive_name="$mi_dir_name-$pos-$parch"
+    mi_archive_dir="${buildPath}/${mi_dir_name}"
+    mkdir -p $mi_archive_dir
 
-    cp -r "${baseDir}/server_config.yaml" $zipdir > /dev/null 2>&1
-    cp -r "${baseDir}/LICENSE" $zipdir > /dev/null 2>&1
+    cp -r "${baseDir}/docs/README.md" $mi_archive_dir > /dev/null 2>&1
```
Comment (a5anka) [inline cmd/build.sh]: Shall we add the README.md to root directory instead of in a docs folder similar to License file?

File: cmd/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# WSO2 Micro Integrator Command Line tool (MI-CLI)
```
Comment (a5anka) [inline cmd/docs/README.md]:
```suggestion
# WSO2 Micro Integrator Command Line tool
```

File: cmd/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+View and manage the micro integrator runtime using the “mi” command line tool. Some of the usages of the command line tool include,
```
Comment (a5anka) [inline cmd/docs/README.md]:
```suggestion
You can view and manage the Micro Integrator instances using the “mi” command line tool. Some of the usages of the command line tool include,
```

File: cmd/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+1. Get a list of deployment runtime artifacts
```
Comment (a5anka) [inline cmd/docs/README.md]:
```suggestion
1. Get a list of deployment runtime artifacts.
```

File: cmd/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+2. Inspect details of each runtime artifact such as a proxy service or an API.
+3. Get the invocation endpoint of an artifact
```
Comment (a5anka) [inline cmd/docs/README.md]:
```suggestion
3. Get the invocation endpoint of an artifact.
```

File: cmd/docs/README.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+### Running
+
+Extract the compressed archive generated to the desired location.
```
Comment (a5anka) [inline cmd/docs/README.md]:
```suggestion
1. Extract the compressed archive generated to the desired location.
```

File: cmd/docs/README.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+Add the MI CLI bin folder to PATH in unix-based Operating System (Linux, Solaris and Mac OS X)
```
Comment (a5anka) [inline cmd/docs/README.md]:
```suggestion
2. Add the MI CLI bin folder to PATH in UNIX-based Operating System (Linux, Solaris, and Mac OS X)
```

File: cmd/docs/README.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+`export PATH=/path/to/mi/cli/directory/bin:$PATH`
```
Comment (a5anka) [inline cmd/docs/README.md]:
```suggestion
    `$ export PATH=/path/to/mi/cli/directory/bin:$PATH`
```

File: cmd/docs/README.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+Then execute,
```
Comment (a5anka) [inline cmd/docs/README.md]:
```suggestion
4. Then execute, 
```

File: cmd/docs/README.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+Extract the compressed archive generated to the desired location.
+
+Add the MI CLI bin folder to PATH in unix-based Operating System (Linux, Solaris and Mac OS X)
+
+`export PATH=/path/to/mi/cli/directory/bin:$PATH`
+
+Then execute, 
+
+`$ mi`
```
Comment (a5anka) [inline cmd/docs/README.md]:
```suggestion
    `$ mi`
```

File: cmd/docs/README.md [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+Execute mi --help for further instructions.
+
+### Add Command Autocompletion
+
+`$ source /path/to/mi/cli/directory/mi_bash_completion.sh`
```
Comment (a5anka) [inline cmd/docs/README.md]: Shall we also mention that this is only available for UNIX based operating systems?

File: cmd/docs/README.md [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+
+### Configuration
+
+##### Q. How to Enable the Management API
```
Comment (a5anka) [inline cmd/docs/README.md]:
```suggestion
##### How to Enable the Management API
```

File: cmd/docs/README.md [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+
+By default the Management API is disabled. To use the Management API you must use the system property `-DenableManagementApi` when starting the micro integrator
+
+**NOTE: These APIs are not protected, do not enable this in production. Take extra measures to secure this port if you are enabling this in production.**
+
+##### Q. Management API Address and Port
```
Comment (a5anka) [inline cmd/docs/README.md]:
```suggestion
##### Management API Address and Port
```

File: cmd/docs/README.md [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+
+`$ source /path/to/mi/cli/directory/mi_bash_completion.sh`
+
+### Configuration
+
+##### Q. How to Enable the Management API
+
+By default the Management API is disabled. To use the Management API you must use the system property `-DenableManagementApi` when starting the micro integrator
+
+**NOTE: These APIs are not protected, do not enable this in production. Take extra measures to secure this port if you are enabling this in production.**
```
Comment (a5anka) [inline cmd/docs/README.md]:
```suggestion
**NOTE: These APIs are not protected using an authorization mechanism. Therefore take extra measures to secure this port if you are enabling this in production.**
```

File: cmd/docs/README.md [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+
+##### Q. Management API Address and Port
+
+To configure the address and the port of the Management API in the CLI use the [**init**](#init) command. This will generate a file called server_config.yaml which contains the address and the port. If the init command was not used, the address and the port will have the default values
+
+NOTE: The default address is https://localhost and the port is 9164
```
Comment (a5anka) [inline cmd/docs/README.md]:
```suggestion
NOTE: The default hostname is localhost and the port is 9164.
```

File: cmd/build.sh (hunk 1 above)
Comment (SanojPunchihewa) [reply cmd/build.sh]: In the distribution, it's added to the root directory
.
├── bin
│   └── mi
└── LICENSE
└── README.md
└── mi_bash_completion.sh

File: cmd/docs/README.md (hunk 6 above)
Comment (SanojPunchihewa) [reply cmd/docs/README.md]: Since we pack the README inside the archive, this step was removed in cdbd7df

**Pull request merged**